api_types.ts

```diff
@@ -805,6 +807,63 @@ export interface Logger {
   error(message: string, ...args: LoggerParamType[]): void;
 }
 
+export enum InvocationErrorType {
+  Timeout = 'Timeout',
+
+  ResponseTooLarge = 'ResponseTooLarge',
+
+  HttpStatusError = 'HttpStatusError',
+
+  /**
+   * Could mean 3rd party API rate limit or a rate limit imposed by Coda.
+   */
+  RateLimitExceeded = 'RateLimitExceeded',
+
+  Unknown = 'Unknown',
+}
+
+interface BaseInvocationError {
+  type: InvocationErrorType;
+  message: string;
+  errorData?: object;
```

Comment:
maybe this is better as a string, with the understanding that it may or may not be json?